web/packages/teleport/src/Roles/RoleEditor/RoleEditorAdapter.tsx

```diff
@@ -68,6 +69,13 @@ export function RoleEditorAdapter({
     convertToRole(originalContent);
   }, [originalContent]);
 
+  const onRoleUpdate = useCallback(
+    debounce((role: Role) => {
+      roleDiffProps.updateRoleDiff(role);
+    }, 500),
```

Comment:
This 500 here has a meaning that is worth naming explicitly. Consider extracting it to a constant.

Reply:
Also: since it's not our goal to set the `this` pointer inside `updateRoleDiff` to `roleDiffProps`, a simpler construct may do the job equally well: `debounce(roleDiffProps.updateRoleDiff, 500)`.